v1/tarball/image_test.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tarball
+
+import (
+	"archive/tar"
+	"bytes"
+	"io"
+	"reflect"
+	"testing"
+
+	"github.com/google/go-containerregistry/v1"
+)
+
+var testManifest = []byte(`{
+"config": {
+	"digest": "sha256:deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef"
+}
+}`)
+
+func createTarReader(t *testing.T, files map[string][]byte) io.Reader {
+	var buf bytes.Buffer
+	w := tar.NewWriter(&buf)
+	defer w.Close()
+	for path, contents := range files {
+		hdr := tar.Header{
+			Name: path,
+			Size: int64(len(contents)),
+			Mode: tar.TypeReg,
+		}
+		if err := w.WriteHeader(&hdr); err != nil {
+			t.Fatalf("Error writing header to test tarfile: %v", err)
+		}
+		if _, err := w.Write(contents); err != nil {
+			t.Fatalf("Error writing contents to test tarfile: %v", err)
+		}
+	}
+	return &buf
+}
```

Comment:
I'd recommend using `container_image` to produce all of your test data.  It builds *exactly* what you are looking for :)

Reply:
bazel's container_image?